### PR TITLE
[Subworkflow] Registration subworkflow.

### PR DESCRIPTION
**Pull request by gagnonanthony**
## Filling your PR description

Registration subworkflow with relevant tests.

### CODE REVIEW

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Super work ! I'd like more documentation coming with the subworkflows, so it's pretty much all the comments I have :+1:

File: subworkflows/nf-scil/registration/meta.yml
[the remark this answers is not in the document]
Comment (gagnonanthony) [reply subworkflows/nf-scil/registration/meta.yml:1]: I added additional descriptions, tell me if you think it is sufficient!

**Comment by gagnonanthony** [comment]
For some reasons, it seems like the nf-test setup action is acting out again 🤔 @AlexVCaron

**Comment by AlexVCaron** [comment]
It should be fixed on the `main`. I needed to modify the fork for PR on the official action, I changed the version we use right now to target the commit before I started that. Sorry for that :stuck_out_tongue:

**Comment by AlexVCaron** [comment]
@gagnonanthony Sync yourself with the main, the `github workflows` should now be fixed !

File: subworkflows/nf-scil/registration/main.nf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        else {
+            // ** Set up input channel, input are inversed compared to REGISTER_ANATTODWI. ** //
+            if ( ch_mask ) {
+                ch_register = ch_ref.combine(ch_image, by: 0)
+                                    .combine(ch_mask, by: 0)
+            }
+            else {
+                ch_register = ch_ref.combine(ch_image, by: 0)
+                                    .map{ [it[0], it[1], it[2], []] }
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/main.nf:43]:
```suggestion
                                    .map{ it + [[]] }
```

File: subworkflows/nf-scil/registration/meta.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+name: "registration"
+description: |
+  Subworkflow to perform registration between a moving and a fixed image (e.g. T1 -> DWI).
+  It requires as input at least a moving (ch_image) and a reference image (ch_ref) to properly
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/meta.yml:4]:
```suggestion
  It requires as input at least a moving (ch_image) and a reference (ch_ref) image to properly
```

File: subworkflows/nf-scil/registration/meta.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  perform registration. Two modes are available:
+  1)  if a metric file is supplied (ch_metric), the subworkflow will use the REGISTER_ANATTODWI
+      module calling AntsRegistration and
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/meta.yml:7]:
```suggestion
      module calling AntsRegistration, with the metric as additional target
```

File: subworkflows/nf-scil/registration/meta.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  2)  if a metric file is NOT supplied, the subworkflow will use the REGISTRATION_ANTS module calling
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/meta.yml:8]:
```suggestion
  2)  if NO metric file is supplied, the subworkflow will use the REGISTRATION_ANTS module calling
```

File: subworkflows/nf-scil/registration/meta.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      antsRegistrationSyN.sh or antsRegistrationSyNQuick.sh.
+  Typical next steps after this subworkflow would be ANATOMICAL_SEGMENTATION or TRACKING.
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/meta.yml:10]: Here, I was more talking about what to do with the transformations computed and how to use the `ANTSAPPLYTRANSFORMS` module on images or tractograms.

File: subworkflows/nf-scil/registration/main.nf [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            // ** Setting outputs ** //
+            image_warped = REGISTRATION_ANTS.out.image
+            transfo_image = REGISTRATION_ANTS.out.transfo_image
+            transfo_trk = REGISTRATION_ANTS.out.transfo_trk
+        }
+
+    emit:
+        image_warped  = image_warped           // channel: [ val(meta), [ image ] ]
+        transfo_image = transfo_image          // channel: [ val(meta), [ warp, affine ] ]
+        transfo_trk   = transfo_trk            // channel: [ val(meta), [ inversewarp, inverseaffine ] ]
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/main.nf:61]: This is inverted (or we did something wrong in the registration modules). The correct order should be `[val(meta), [inverseAffine, inverseWarp] ]`

File: subworkflows/nf-scil/registration/main.nf [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            REGISTER_ANATTODWI ( ch_register )
+            ch_versions = ch_versions.mix(REGISTER_ANATTODWI.out.versions.first())
+
+            // ** Setting outputs ** //
+            image_warped = REGISTER_ANATTODWI.out.t1_warped
+            transfo_image = REGISTER_ANATTODWI.out.transfo_image
+            transfo_trk = REGISTER_ANATTODWI.out.transfo_trk
+        }
+        else {
+            // ** Set up input channel, input are inversed compared to REGISTER_ANATTODWI. ** //
```
Comment (AlexVCaron) [inline subworkflows/nf-scil/registration/main.nf:36]:
```suggestion
            // ** Set up input channel, input are inverted compared to REGISTER_ANATTODWI. ** //
```

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Last few comments, then I think it's ready for merge ! :boat:

**Review by AlexVCaron — APPROVED** [review]
All good on my end ! AFter you remove WIP and @ThoumyreStanislas passes through the PR, LGTM !

**Pull request merged**